src/QsCompiler/Transformations/FunctorGeneration.cs

```diff
@@ -15,13 +15,17 @@
 
 namespace Microsoft.Quantum.QsCompiler.Transformations.FunctorGeneration
 {
+    using ExpressionKind = QsExpressionKind<TypedExpression, Identifier, ResolvedType>;
+    using ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+    using TypeArgsResolution = ImmutableArray<Tuple<QsQualifiedName, NonNullable<string>, ResolvedType>>;
+
     /// <summary>
     /// Scope transformation that replaces each operation call within a given scope
     /// with a call to the operation after application of the functor given on initialization.
     /// The default values used for auto-generation will be used for the additional functor arguments.
     /// Additional functor arguments will be added to the list of defined variables for each scope.
     /// </summary>
-    public class ApplyFunctorToOperationCalls
+    internal class ApplyFunctorToOperationCalls
```

Comment:
Please check with @bettinaheim about changing the access modifiers for the existing transformations from public to internal.

Reply:
True... while we don't expect anything else to be using them, it is technically a change in the API.

Reply:
Let's properly deprecate things when we do that. In this case, while it seems like it is no longer in use by anything outside the compiler, my inclination would be to keep it as part of the public API. Or do you see any reason why that particular piece would likely change in the future? Same for IgnoreOuterBlockInConjugations<T>.
As far as AddVariableDeclarations<T> is concerned, it may make sense to properly deprecate it here, and add an internal class to the BasicTransformations namespace instead.

Reply:
I'll put `ApplyFunctorToOperationCalls` and `IgnoreOuterBlockInConjugations` back to public. I think I might need a little more clarification on how to properly deprecate `AddVariableDeclarations`; do I add a decorator on this one and copy the implementation to the internal replacement? I'm fine with just putting this one back to public and leaving the API surface as is until we want to do a more formal review of the compiler public API, if that would be more straightforward for now.

Reply:
The suggestion is to copy the implementation for AddVariableDeclarations to the BasicTransformations (it can be kept internal), have this one here just inherit from the one in BasicTransformations to keep it functional, and decorate it with the Obsolete attribute.

Reply:
Unfortunately, I can't have the one in BasicTransformations be internal while having one here inherit and be public, because a public class cannot inherit from an internal class:
`FunctorGeneration.cs(116,18): error CS0060: Inconsistent accessibility: base class 'AddVariableDeclarations<T>' is less accessible than class 'AddVariableDeclarations<T>'`
So they either both have to be public or both have to be internal, and once they have the same accessibility it seems to defeat the point of duplicating it, since we did that just to deprecate the public one. Should I just have two copies of the same implementation?

Reply:
Ah, right, that's unfortunate. I guess duplicating will have to do if we want to remove it from the public API. I think that should be fine. I'll create a task to make sure we are not forgetting about removing it.